cmd/resource/status/cmd/helpers_test.go

```diff
@@ -0,0 +1,279 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TableOutput contains information about the output
+// from a command after it has been parsed. The output
+// can contain one or more frames, each being a version
+// of the table output. Rows that has an unknown format
+// are captured separately.
+type TableOutput struct {
+	Frames      []TableOutputFrame
+	UnknownRows []string
+}
+
+// allAggStatuses returns all the aggregated statuses
+// found in the table output. They will be in chronological order.
+func (t *TableOutput) allAggStatuses() []string {
+	var statuses []string
+	for _, frame := range t.Frames {
+		if frame.AggregateStatus != "" {
+			statuses = append(statuses, frame.AggregateStatus)
+		}
+	}
+	return statuses
+}
+
+// dedupedAggStatuses returns the aggregated statuses
+// from the table output, but it removes all duplicates, so
+// only the transition between statuses are captured.
+func (t *TableOutput) dedupedAggStatuses() []string {
+	var dedupedStatuses []string
+	statuses := t.allAggStatuses()
+	var previousStatus string
+	for _, s := range statuses {
+		if s != previousStatus {
+			dedupedStatuses = append(dedupedStatuses, s)
+			previousStatus = s
+		}
+	}
+	return dedupedStatuses
+}
+
+// resources returns all the resources found in
+// the table output.
+func (t *TableOutput) resources() []ResourceIdentifier {
+	seenResources := make(map[ResourceIdentifier]bool)
+	var resources []ResourceIdentifier
+	for _, frame := range t.Frames {
+		for _, resource := range frame.Resources {
+			r := ResourceIdentifier{
+				apiVersion: resource.apiVersion,
+				kind:       resource.kind,
+				namespace:  resource.namespace,
+				name:       resource.name,
+			}
+			_, found := seenResources[r]
+			if !found {
+				seenResources[r] = true
+				resources = append(resources, r)
+			}
+		}
+	}
+	return resources
+}
+
+// dedupedStatusesForResource returns the deduped statuses for the given
+// resource. Deduped means that we only keep statuses from frames where the
+// status changed from the previous frame.
+func (t *TableOutput) dedupedStatusesForResource(resource ResourceIdentifier) []string {
+	var dedupedStatuses []string
+	var previousStatus string
+	for _, frame := range t.Frames {
+		for _, r := range frame.Resources {
+			if r.apiVersion == resource.apiVersion &&
+				r.kind == resource.kind &&
+				r.namespace == resource.namespace &&
+				r.name == resource.name {
```

Comment:
would be helpful to have an Equals method on ResourceIdentifier or reuse [ResId](https://github.com/kubernetes-sigs/kustomize/blob/master/api/resid/resid.go) which already has it.

Reply:
I have added an Equals method to ResourceIdentifier. 
I looked into ResId before, but it includes version in the identifier which I don't think I want. Any reason why it builds on GroupVersionKind instead of only GroupKind?